### DISCUSSION TITLE
Dropping support for Python 3.7

### DISCUSSION THREAD

**Discussion by RhetTbull** [RhetTbull/osxphotos#447] [General]
osxphotos currently works with python 3.7+.  There are some features of 3.8 I'd like to use  -- e.g. `@functools.cached_property` that would allow me to really simplify some of the code.  I plan to soon drop support for python 3.7.  If this is a big limitation for you, please discuss here.

**Comment by neilpa** [comment]
One downside is that it appears that system Python for Big Sur is still 3.7.

```
$ sw_vers 
ProductName:	macOS
ProductVersion:	11.4
BuildVersion:	20F71

$ /usr/bin/python3 --version
Python 3.7.3
```

I'm not sure how many users depend on that vs an updated version via `homebrew`, `virtualenv`, etc.

**Reply by RhetTbull** [reply]
Thanks--good point.  I'm surprised Big Sur still has a system python...I thought Apple was [dropping pre-installed python](https://www.reddit.com/r/apple/comments/bwulrh/apple_will_soon_be_removing_python_perl_and_ruby/).  It's not urgent but I do eventually want to do a big refactor of `PhotoInfo` which includes making use of `@functools.cached_property`

**Comment by RhetTbull** [comment]
As of release 0.47.0 I've dropped support for python 3.7.  There were features I wanted to use in 3.8 and I figure that supporting the last 3 releases of Python (3.8, 3.9, 3.10) is sufficient.  (python 3.11 is in alpha now)  It's very easy to install python from python.org or just use brew to install pipx then `pipx install osxphotos` without messing with the system python.
